src/binary-finder.ts

```diff
@@ -85,46 +64,42 @@ const vsCodeSettingsStrategy: LocatorStrategy = {
 				debug(
 					"Biome binary path is empty in VS Code settings, skipping",
 				);
-				return undefined;
+				return;
 			}
 
-			const resolvedBinPath = Utils.resolvePath(path, bin).toString();
+			const resolvedBinPath = path
+				? Utils.resolvePath(path, bin).toString()
+				: bin;
 
-			const biome = Uri.file(resolvedBinPath);
+			const biome = Uri.parse(resolvedBinPath);
 
 			if (await fileExists(biome)) {
 				return biome;
 			}
-
-			return undefined;
 		};
 
 		const findPlatformSpecificBinary = async (
 			bin: Record<string, string>,
 		): Promise<Uri | undefined> => {
 			debug(
 				"Trying to find platform-specific Biome binary in VS Code settings",
-				{
-					bin,
-				},
+				{ bin },
 			);
 
 			if (platformIdentifier in bin) {
-				return findBinary(bin[platformIdentifier]);
+				return await findBinary(bin[platformIdentifier]);
```

Comment:
We actually returned a `Promise` from an `async` function. This would've resulted in a `Promise<Promise<T>>`.

Reply:
good catch, thanks !